onnxruntime/core/optimizer/reshape_fusion.cc

```diff
@@ -83,42 +78,42 @@ static bool DistilBert_Check(Graph& graph, const Node& concat, const Node& layer
 
   std::vector<graph_utils::EdgeEndToMatch> linear_path{
     {0, 0, "Add", {7}, kOnnxDomain},
-    {0, 0, "MatMul", {1, 9}, kOnnxDomain},
-    {0, 0, "Add", {7}, kOnnxDomain}};
+    {0, 0, "MatMul", {1, 9}, kOnnxDomain}};
   if (!graph_utils::FindPath(reshape, true, linear_path, edges, logger)) {
     return false;
   }
-  const Node& linear_path_root_add = edges[2]->GetNode();
-  if (linear_path_root_add.Index() != layernorm_add.Index()) {
-    return false;
-  }
+
   const Node& linear_path_add = edges[0]->GetNode();
   const Node& linear_path_matmul = edges[1]->GetNode();
 
-  if (layernorm_add.InputDefs().size() < 2) {
+  const Node& node_before_matmul = *graph_utils::GetInputNode(linear_path_matmul, 0);
+  if (node_before_matmul.Index() != root.Index()) {
     return false;
   }
-  const NodeArg& layernorm_add_b = *(layernorm_add.InputDefs()[1]);
-  if (!graph_utils::IsInitializer(graph, layernorm_add_b.Name(), true)) {
+
+  if (linear_path_add.InputDefs().size() < 2) {
+    return false;
+  }
+  const NodeArg& linear_path_add_b = *(linear_path_add.InputDefs()[1]);
+  if (!graph_utils::IsInitializer(graph, linear_path_add_b.Name(), true)) {
     return false;
   }
 
-  const ONNX_NAMESPACE::TensorShapeProto* layernorm_add_b_shape = layernorm_add_b.Shape();
-  if (layernorm_add_b_shape == nullptr || layernorm_add_b_shape->dim_size() != 1) {
+  const ONNX_NAMESPACE::TensorShapeProto* linear_path_add_b_shape = linear_path_add_b.Shape();
+  if (linear_path_add_b_shape == nullptr || linear_path_add_b_shape->dim_size() != 1) {
     return false;
   }
 
-  const ONNX_NAMESPACE::TensorShapeProto_Dimension& layernorm_add_b_shape_dim = layernorm_add_b_shape->dim(0);
-  if (!utils::HasDimValue(layernorm_add_b_shape_dim)) {
+  const ONNX_NAMESPACE::TensorShapeProto_Dimension& linear_path_add_b_shape_dim = linear_path_add_b_shape->dim(0);
+  if (!utils::HasDimValue(linear_path_add_b_shape_dim)) {
     return false;
   }
-  int64_t hidden_size = layernorm_add_b_shape_dim.dim_value();
+  int64_t hidden_size = linear_path_add_b_shape_dim.dim_value();
 
-  if (linear_path_add.InputDefs().size() < 2 || linear_path_matmul.InputDefs().size() < 2) {
+  if (linear_path_matmul.InputDefs().size() < 2) {
```

Comment:
nit. I think it is not needed. MatMul shall have two inputs, otherwise the graph is not valid.